### DISCUSSION TITLE
Terrazzo 2.0 Roadmap

### DISCUSSION THREAD

**Discussion by drwpow** [terrazzoapp/terrazzo#288] [Announcements]
In the past couple months, this repo has been renamed to `terrazzoapp/terrazzo`, and the `@terrazzo` packages are publicly-available on npm for trial. But what’s the plan? When will Terrazzo be stable?

## Terrazzo 2.0

Terrazzo will become stable at `2.0` (not `1.0`) because it’s really a continuation of Cobalt! (Likewise, Cobalt will get some npm updates that suggest to start using `@terrazzo/*` packages instead).

## TODOs

The following features are still big TODOs that are blocking the stable version. There’s too much completed work to list out here that are all improvements over Cobalt (read the changelog). Here are the only items remaining:

- [x] ~~**Multiple tokens file support.**~~ Just done!
- [x] ~~**YAML support.**~~ Done!
- [ ] **Linting 2.0.** The linter shipped with Cobalt lacks an AST, which is one of the most important parts of linting. So this means the linting engine still has to be rebuilt to support this. This is the only remaining thing to do in the core engine.
- [ ] **Tailwind plugin.** Not a huge lift, but Cobalt’s Tailwind plugin was still rudimentary, and not usable in a lot of production setups. This should get a little more TLC than Cobalt’s plugin.

And that’s pretty much it! As mentioned, be sure to check out the changelog for the comprehensive list of all that’s improved.

But it’s not just open work to complete, it’s also…

## DTCG 1.0

As of 2024, I’ve been a contributor to the DTCG spec, and it’s shooting for a stable release by end of year! 🎉 This would be a perfect opportunity to release Terrazzo when that’s ready (or, if not 100% finalized, when it’s > 90% done and not expecting any last-minute shifts).

## 🧪 Bonus: Token Lab

Something that’s been teased for years, both on the original Cobalt site, and issues, are some design system tools to build tokens. These will also get released later this year (finally), near the Terrazzo 2.0 stable release (maybe not _blocking_ the 2.0 release depending on timing, but won’t trail far behind).

A WebGL-powered, CSS Color Module 4-supporting [React Color Picker](https://terrazzo.app/docs/components/color-picker/) is just a small taste of what’s to come 🌈.

**Comment by davejsdev** [comment]
@drwpow So far all the components in Token Lab are custom built and use CSS modules.

Is that an intentional decision? Would you be opposed to introducing shadcn/ui components and adding Tailwind?

**Reply by drwpow** [reply]
I’m opposed to Tailwind because it’s an overhead abstraction. I’d much rather have a cleaner 1:1 mapping of tokens → generic CSS because it can appeal to more scenarios (also I would imagine more companies that are using heavy design token setups likely aren’t using prebuilt UI kits).

I think using some prebuilt component library would be a good idea as long as it’s on the more themeable end. I don’t have a preference though. [React Aria](https://uikits.design) might be a good candidate. I recently found [UI Kits](https://uikits.design) which seems to be an up to date listing

**Reply by davejsdev** [reply]
>  I’d much rather have a cleaner 1:1 mapping of tokens → generic CSS because it can appeal to more scenarios (also I would imagine more companies that are using heavy design token setups likely aren’t using prebuilt UI kits).

So you mean the choice to keep more raw CSS is intended to attract more design engineers (who are less likely to know Tailwind) to contribute to the token lab?
